src/main/groovy/com/bugsnag/android/gradle/BugsnagUploadNdkTask.groovy

```diff
@@ -0,0 +1,155 @@
+    String variantName
+    File projectDir
+    File rootDir
+    String toolchain
+
+    BugsnagUploadNdkTask() {
+        super()
+        this.description = "Generates and uploads the NDK mapping file(s) to Bugsnag"
+    }
+
+    @TaskAction
+    def upload() {
+
+        if (intermediatePath == null) {
+            return
+        }
+
+        // Look for the shared objects in likely folders
+        File binariesFile = null;
+        for (File intDir : intermediatePath.listFiles()) {
+            if (intDir.name.equals("cmake")) { // CMake
+                binariesFile = intDir;
+            } else if (intDir.name.equals("binaries")) { // Experimental
+                binariesFile = intDir;
+            } // TODO: where do the build-ndk files get output?
+        }
+
+        if (binariesFile == null) {
+            project.logger.error("Unable to locate correct path to intermediate binaries under " + intermediatePath.absolutePath)
+            return
+        }
+
+        // Read the API key and Build ID etc..
+        super.readManifestFile();
+
+        // Create the files to upload
+        File intermediateBinaries = new File(binariesFile.absolutePath + File.separator + variantName + File.separator + "obj")
+
+        for (File archDir : intermediateBinaries.listFiles()) {
+            if (archDir.isDirectory()) {
+                String arch = archDir.getName();
+
+                for (File sharedObject : archDir.listFiles(new SharedObjectFilter())) {
+                    File outputFile = createSymbolsForSharedObject(sharedObject, arch)
+
+                    if (outputFile != null) {
+                        uploadSymbols(outputFile, arch, sharedObject.getName())
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Class to filter files to just contain shared objects
+     */
+    private static class SharedObjectFilter implements FilenameFilter {
+        @Override
+        public boolean accept(File dir, String name) {
+            return name.endsWith(".so");
+        }
+    }
+
+    /**
+     * Uses objdump to create a symbols file for the given shared object file
+     * @param sharedObject the shared object file
+     * @param arch the arch of the file
+     * @return the output file location, or null on error
+     */
+    private File createSymbolsForSharedObject(File sharedObject, String arch) {
+        // Get the path the version of objdump to use to get symbols
+        File objDumpPath = getObjDumpExecutable(arch)
+        if (objDumpPath != null) {
+
+            try {
+                File outputFile = new File(symbolPath.getAbsolutePath() + File.separator + arch + ".txt");
+
+                // Call objdump, redirecting output to the output file
+                ProcessBuilder builder = new ProcessBuilder(objDumpPath.toString(), "--disassemble", "--demangle", "--line-numbers", "--section=.text", sharedObject.toString())
+                builder.redirectOutput(outputFile)
```

Comment:
> I'm not sure how much work we want to do here

Enough to not upload the entire disassembled library. First and last are at least significantly less and shouldn't require much more work.